packages/connection-form/src/components/connection-form.tsx

```diff
@@ -97,6 +109,165 @@ type ConnectionFormPropsWithoutPreferences = {
+  marginRight: spacing[2],
+});
+
+type ColorCircleGlyphProps = { hexColor?: string };
+const ColorCircleGlyph = createGlyphComponent(
+  'ColorCircle',
+  ({ hexColor, ...props }: any & ColorCircleGlyphProps) => (
+    <svg
+      {...props}
+      className={colorPreviewStyles}
+      width="16"
+      height="16"
+      viewBox="0 0 16 16"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <g clipPath="url(#clip0_756_18092)">
+        <line
+          opacity="0.5"
+          x1="16.3536"
+          y1="-0.146447"
+          x2="0.353554"
+          y2="15.8536"
+          stroke="#DB3030"
+        />
+      </g>
+      <rect
+        x="0.5"
+        y="0.5"
+        width="15"
+        height="15"
+        rx="7.5"
+        stroke="#889397"
+        fill={hexColor}
+      />
+      <defs>
+        <clipPath id="clip0_756_18092">
+          <rect width="16" height="16" rx="8" fill="white" />
+        </clipPath>
+      </defs>
+    </svg>
+  )
+) as React.FunctionComponent<ColorCircleGlyphProps>;
+
+const personalizationSectionLayoutStyles = css({
+  display: 'grid',
+  gridTemplateColumns: '1fr 1fr',
+  gridTemplateRows: 'auto',
+  gridTemplateAreas: `
+    'name-input color-input'
+    'favorite-marker favorite-marker'
+  `,
+  gap: spacing[4],
+  marginBottom: spacing[4],
+});
+
+type ConnectionPersonalizationFormProps = {
+  personalizationOptions: ConnectionPersonalizationOptions;
+  updateConnectionFormField: UpdateConnectionFormField;
+};
+
+function ConnectionPersonalizationForm({
+  updateConnectionFormField,
+  personalizationOptions,
+}: ConnectionPersonalizationFormProps): React.ReactElement {
+  const showFavoriteActions = useConnectionFormPreference(
+    'showFavoriteActions'
+  );
+
+  const onChangeName = useCallback(
+    (ev: React.ChangeEvent<HTMLInputElement>) => {
+      updateConnectionFormField({
+        type: 'update-connection-personalization',
+        ...personalizationOptions,
+        name: ev.target.value,
+        isNameDirty: true,
+      });
+    },
+    [updateConnectionFormField, personalizationOptions]
+  );
+
+  const onChangeColor = useCallback(
+    (newValue: string) => {
+      updateConnectionFormField({
+        type: 'update-connection-personalization',
+        ...personalizationOptions,
+        color: newValue,
+      });
+    },
+    [updateConnectionFormField, personalizationOptions]
+  );
+
+  const onChangeFavorite = useCallback(
+    (ev: React.ChangeEvent<HTMLInputElement>) => {
+      updateConnectionFormField({
+        type: 'update-connection-personalization',
+        ...personalizationOptions,
+        isFavorite: ev.target.checked,
+      });
+    },
+    [updateConnectionFormField, personalizationOptions]
+  );
+
+  const { connectionColorToHex, connectionColorToName, connectionColorCodes } =
+    useConnectionColor();
+
+  return (
+    <div className={personalizationSectionLayoutStyles}>
+      <TextInput
+        style={{ gridArea: 'name-input' }}
```

Comment:
Nit: this is not really dynamic in any way and so can be defined staically with css and passed as a class name